Userland/Libraries/LibWeb/CSS/CSSStyleRule.cpp

```diff
@@ -102,8 +104,14 @@ void CSSStyleRule::set_selector_text(StringView selector_text)
     auto parsed_selectors = parse_selector(Parser::ParsingContext { realm() }, selector_text);
 
     // 2. If the algorithm returns a non-null value replace the associated group of selectors with the returned value.
-    if (parsed_selectors.has_value())
+    if (parsed_selectors.has_value()) {
         m_selectors = parsed_selectors.release_value();
+        if (auto* window = dynamic_cast<HTML::Window*>(&HTML::current_global_object()); window) {
+            auto& document = window->associated_document();
+            document.style_computer().invalidate_rule_cache();
+            document.invalidate_style();
+        }
```

Comment:
Hmm, what if I'm manipulating the `selectorText` of a style rule inside an iframe?

This feels like it should find its way to the document in a more precise way, perhaps via something like rule -> parent style sheet -> document it's attached to.

Reply:
Good point! I've also added a `document.isConnected()` check here.